client/bookmarkUtil.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+/**
+ * Returns a base bookmark order based on deviceId and platform.
+ * @param {String} deviceId
+ * @param {String} platform
+ * @returns {String}
+ */
+module.exports.getBaseBookmarksOrder = (deviceId, platform) => {
+  return `${(platform === 'ios' || platform === 'android') ? 2 : 1}.${deviceId}.`
+}
+
+/**
+ * Helper to get next order number.
+ * @param {number} lastNumber - last digit of prev order
+ * @param {string} order - the order string
+ * @return {string}
+ */
+const getNextOrderFromPrevOrder = (lastNumber, order) => {
+  if (lastNumber <= 0) {
+    throw new Error('Invalid input order')
+  } else {
+    return order + (lastNumber + 1)
+  }
+}
+
+/**
+ * Helper to get previous order number.
+ * @param {number} lastNumber - last digit of next order
+ * @param {string} order - the order string
+ * @return {string}
+ */
+const getPrevOrderFromNextOrder = (lastNumber, order) => {
+  if (lastNumber <= 0) {
+    throw new Error('Invalid input order')
+  } else if (lastNumber === 1) {
+    return order + '0.1'
+  } else {
+    return order + (lastNumber - 1)
+  }
+}
+
+/**
+ * Returns current bookmark order based on previous and next bookmark order.
+ * @param {String} prevOrder
+ * @param {String} nextOrder
+ * @returns {String}
+ */
+module.exports.getBookmarkOrder = (prevOrder, nextOrder) => {
+  let prevOrderSplit = prevOrder.split('.')
+  let nextOrderSplit = nextOrder.split('.')
+  if (prevOrderSplit.length === 1 && nextOrderSplit.length === 1) {
+    throw new Error(`Invalid previous and next orders: ${prevOrderSplit} and ${nextOrderSplit}`)
+  }
+  let order = ''
+  if (nextOrderSplit.length === 1) {
+    // Next order is an empty string
+    if (prevOrderSplit.length > 2) {
+      for (var i = 0; i < prevOrderSplit.length - 1; i++) {
+        order += prevOrderSplit[i] + '.'
+      }
+      let lastNumber = parseInt(prevOrderSplit[prevOrderSplit.length - 1])
+      order = getNextOrderFromPrevOrder(lastNumber,  order)
+    }
+  } else if (prevOrderSplit.length === 1) {
+    // Prev order is an empty string
+    if (nextOrderSplit.length > 2) {
```

Comment:
can replace this for loop with `order = nextOrder + '.'`

Reply:
the same as above